crates/holochain_types/src/app/app_manifest.rs

```diff
@@ -0,0 +1,63 @@
+#![warn(missing_docs)]
+
+//! Defines the hApp Manifest YAML format, including validation.
+
+use mr_bundle::{Location, Manifest};
+use std::path::PathBuf;
+
+pub(crate) mod app_manifest_v1;
+pub mod app_manifest_validated;
+mod current;
+mod error;
+
+pub use current::*;
+pub use error::*;
+
+use self::{app_manifest_validated::AppManifestValidated, error::AppManifestResult};
+use app_manifest_v1::AppManifestV1;
+
+/// Container struct which uses the `manifest_version` field to determine
+/// which manifest version to deserialize to.
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize, derive_more::From)]
+#[serde(tag = "manifest_version")]
+#[allow(missing_docs)]
+pub enum AppManifest {
+    #[serde(rename = "1")]
+    V1(AppManifestV1),
+}
+
+impl Manifest for AppManifest {
+    fn locations(&self) -> Vec<Location> {
+        match self {
+            AppManifest::V1(m) => m
+                .slots
+                .iter()
+                .filter_map(|slot| slot.dna.location.clone())
+                .collect(),
+        }
+    }
+
+    fn path() -> PathBuf {
+        "happ.yaml".into()
```

Comment:
`yaml` or `yml`? I feel like `yml` is a little more common, but either way we should be consistent with conductor config etc..

Reply:
I was about to agree with you on that and rename stuff, but then found the official line at https://yaml.org/faq.html:

> Please use ".yaml" when possible.

I'm pretty indifferent, leaning towards sticking with `.yaml`. We do use `toml` and not `tml` after all. We could get fancy and support both `yaml` and `yml` too.

Also I guess it's kind of low-key insulting to drop the "ain't" from the acronym, making it "Yaml's Markup Language" ;)

Reply:
I'm cool with `yaml`, but we should change the conductor config extension, then : )

Reply:
per discussion today: `yaml` it is